### ISSUE TITLE
Search by _id, the default Modifier using exact

### ISSUE THREAD

**Issue by whyfate** [FirelyTeam/spark#293]
**Is your feature request related to a problem? Please describe.**
When I use `_id=1` to query a resource, I want the default modifier to be exact instead of requiring me to specify it `_id:exact=1`.

**Comment by whyfate** [comment]
I think it's more friendly.

**Comment by brianpos** [comment]
The _id search parameter is of type reference, which is implicitly an exact search, so the exact isn't needed.

**Comment by whyfate** [comment]
> The _id search parameter is of type reference, which is implicitly an exact search, so the exact isn't needed.

Yes,I think so.So how to do it better?
https://github.com/FirelyTeam/spark/blob/r4/master/src/Spark.Mongo/Search/Searcher/CriteriaMongoExtensions.cs#L99
```csharp
              if (parameterName == "_id")
              {
                    parameterName = "fhir_id"; //See MongoIndexMapper for counterpart.
                    // default Modifier value is exact when parameterName is _id. 
                    modifier = string.IsNullOrWhiteSpace(modifier) ? Modifier.EXACT : modifier;
                }
```

**Comment by brianpos** [comment]
Sorry, I thought I was commenting on the fhir-net-api client. You're looking how to fix this in the spark mongo storage engine. And yes it should be done.
I'll take a quick peek to see if I can spot something, but probably better for others to do...

**Comment by whyfate** [comment]
So do you think it's better to deal with it in fhir-net-api?

**Comment by brianpos** [comment]
No, the Spark Mongo engine is the correct place to deal with this, it's a server issue not a caller issue.

**Comment by whyfate** [comment]
That's right,I think so.

**Comment by andy-a-o** [comment]
Fixed in `master` branch for DSTU2. About to merge it to `stu3/master` and `r4/master`.

**Comment by andy-a-o** [comment]
https://github.com/FirelyTeam/spark/blob/340c7b278a24bbe1366831515522c80f145e9ca3/src/Spark.Mongo/Search/Searcher/CriteriaMongoExtensions.cs#L105

**Comment by kennethmyhra** [comment]
Fixed in version 1.5.3
